models/workitemtype_cache_blackbox_test.go

```diff
@@ -0,0 +1,64 @@
+package models_test
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+var cache = models.NewWorkItemTypeCache()
+
+func TestNotExistingType(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+
+	_, ok := cache.Get("notexists")
+	assert.False(t, ok)
+}
+
+func TestReadingWriting(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+
+	_, ok := cache.Get("type1")
+	assert.False(t, ok)
+
+	wit := models.WorkItemType{}
+	wit.Name = "type1"
+	cache.Put(wit)
+
+	cachedWit, ok := cache.Get("type1")
+	assert.True(t, ok)
+	assert.Equal(t, wit, cachedWit)
+}
+
+func TestConcurrentAccess(t *testing.T) {
```

Comment:
It is not clear to me what you are testing in this test. What possible problem do you want to find?